### PR TITLE
EOS-17997: Create IEM for HA

### PR DESCRIPTION
**Pull request by ajay-paratmandali**
Signed-off-by: Ajay Paratmandali <ajay.paratmandali@seagate.com>

## Problem Statement
<pre>
  <code>
    Story Ref (if any):
    Please add Problem statement here...
EOS-17997
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Please add Problem decription here...
IEM need for node failure and recovery
  </code>
</pre>
## Solution
<pre>
  <code>
    Please add summary of the change,List any dependencies that are required for this change.
IEM
  </code>
</pre>
## Sanity testing on RPM done
<pre>
  <code>
    - No
  </code>
</pre>
## Unit/Manual Testing Description
<pre>
  <code>
    Please describe the tests that you ran to verify your changes.
Testing done on running setup
  </code>
</pre>

### CODE REVIEW

File: low-level/files/iec_mapping/ha [hunk 1]
```diff
@@ -1,4 +1,2 @@
-0080010001,NodeFailure,The cluster has lost one server. System is running in degraded mode.
-0080020001,NodeRecovery,Server has joined back the cluster. System is restored.
-0080030001,IOService,IO services are getting started on current server.
-0080030002,IOService,IO services are stopping on current server due to hardware or service failure.
+0080010001,Node, The cluster has lost $host server. System is running in degraded mode. For more information refer the Troubleshooting guide. Extra Info: host=$host; status=$status;
+0080010002,Node, Server $host has joined back the cluster. System is restored. Extra Info: host=$host; status=$status;
```
[the remark this answers is not in the document]
Comment (ajay-paratmandali) [reply low-level/files/iec_mapping/ha:2]: Yes it is template only message will be replaced with with proper value

[the remark this answers is not in the document]
Comment (indrajitzagade) [reply low-level/files/iec_mapping/ha:2]: @ajay-paratmandali Following value will be copied to "event" field within the IEM alert by SSPL:
"The cluster has lost $host server. System is running in degraded mode. For more information refer the Troubleshooting guide. Extra Info: host=$host; status=$status;"

So we cannot put e.g. $host unless SSPL is equipped to replace these values. Lets put something as we have in R1.

[the remark this answers is not in the document]
Comment (ajay-paratmandali) [reply low-level/files/iec_mapping/ha]: Updated default message similar to R1

**Review by cdeshmukh — APPROVED** [review]
Why are we removing the IEMs btw ?

File: low-level/files/iec_mapping/ha [hunk 2]
```diff
@@ -1,4 +1,2 @@
-0080010001,NodeFailure,The cluster has lost one server. System is running in degraded mode.
-0080020001,NodeRecovery,Server has joined back the cluster. System is restored.
-0080030001,IOService,IO services are getting started on current server.
-0080030002,IOService,IO services are stopping on current server due to hardware or service failure.
+0080010001,Node,The cluster has lost one server. System is running in degraded mode.
```
Comment (indrajitzagade) [inline low-level/files/iec_mapping/ha:1]: One minor suggestion, can we remove "one" from above string? Just put "a" may be.

Comment (ajay-paratmandali) [reply low-level/files/iec_mapping/ha]: Done

**Review by indrajitzagade — APPROVED** [review]

**Review by ajaysrivas — APPROVED** [review]

**Comment by ajay-paratmandali** [comment]
>Why are we removing the IEMs btw ?
In HA R2 IEM code is different than R1 as well as there will be more IEM in future 
as old IEM no longer needed removing it for R2

**Comment by cdeshmukh** [comment]
test this please

**Comment by ajay-paratmandali** [comment]
> test this please

@cdeshmukh 
Tested this by change iem_mapping/ha file on already working system 
as small and config change expecting this should be fast and correct way

**Comment by cdeshmukh** [comment]
> > test this please
> 
> @cdeshmukh
> Tested this by change iem_mapping/ha file on already working system
> as small and config change expecting this should be fast and correct way
> let know if any other testing needed

@ajay-paratmandali  'test this please' comment from admin is a trigger to try mini provisioning again for his PR. But i still see its not triggerred, can you follow up with Gowthamam

**Comment by gowthamchinna** [comment]
test mini-provisioning

**Comment by gowthamchinna** [comment]
test mini-provisioning

**Review by sandeepsgt — APPROVED** [review]

**Review by thavanathan — APPROVED** [review]

**Pull request merged**